Rel8/Internal/Aggregate.hs

```diff
@@ -131,3 +132,11 @@ groupBy (Expr a) = Aggregate Nothing a
 -- | Corresponds to @COUNT(*)@.
 countStar :: Aggregate Int64
 countStar = count (lit @Int64 0)
+
+-- | Corresponds to @COUNT(*) FILTER (WHERE <condition>)@.
+countWhere :: Predicate bool => Expr bool -> Aggregate Int64
+countWhere condition = count (mcase [(condition, lit @Int64 0)])
+
+-- | Corresponds to @SUM(<expr>) FILTER (WHERE <condition>)@.
+sumWhere :: (Predicate bool, DBSum a res, Num a) => Expr bool -> Expr a -> Aggregate res
+sumWhere condition a = sum (case_ [(condition, a)] (lit 0))
```

Comment:
This looks fine, but do you know if PostgreSQL optimises this different to an actual proper `WHERE` on the aggregate function? I don't think Opaleye supports that, but it could be interesting to add to Opaleye in the future.

Reply:
I read in some documentation somewhere that they're equivalent, and the plans look sensible, but I haven't directly verified it.